thunder/torch/__init__.py

```diff
@@ -905,8 +905,10 @@ def pad(a: TensorProxy, /, pad: tuple[int, ...], mode: str | None = "constant",
 
     if value is None:
         value = 0
+    a_typ = to_dtype(a, true_dtype=True)
+    v2 = clang.maybe_convert_to_dtype(value, a_typ)
```

Comment:
`enforce_safe_casting=True`?

Reply:
Sadly, no; the cast is unsafe in PyTorch:
```
>>> x = torch.tensor([5,2])
>>> torch.nn.functional.pad(x, [2,1], value=3.14159)
tensor([3, 3, 5, 2, 3])
```